spotlight-client/src/datastores/Metric.ts

```diff
@@ -0,0 +1,316 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { ValuesType } from "utility-types";
+import { callMetricsApi, DataFile } from "../metricsApi/metricsClient";
+import { NamedEntity } from "./types";
+
+export enum MetricTypes {
+  SentencePopulationCurrent = "SentencePopulationCurrent",
+  SentenceTypesCurrent = "SentenceTypesCurrent",
+}
+
+type MetricsContent = NamedEntity & { methodology: string };
+
+type MetricFactoryOptionsBase = MetricsContent;
+
+type TotalIdentifier = "ALL";
+const TOTAL_KEY: TotalIdentifier = "ALL";
+
+type DemographicView = "total" | "race" | "gender" | "age" | "nofilter";
```

Comment:
What is the difference between the "total" and "nofilter" view?

Reply:
"total" will only let records pass through that match the condition:
```js
{
  raceOrEthnicity: "ALL",
  gender: "ALL", 
  ageBucket: "ALL,
}
```
whereas "nofilter" would just return the entire `Metric.allRecords` array. (I realize now that I did not actually implement the behavior, only the interface definition ... sorry for the confusion!)